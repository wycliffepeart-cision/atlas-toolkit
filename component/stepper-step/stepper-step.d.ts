export declare const STEPPER_STEP_ATTRIBUTES: string[];
export declare class StepperStep extends HTMLElement {
}
//# sourceMappingURL=stepper-step.d.ts.map