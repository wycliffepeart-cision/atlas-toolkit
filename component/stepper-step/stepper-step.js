import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var StepperStepTemplate = "<template>\n  <div part=\"before\"></div>\n  <div part=\"step\">\n    <slot part=\"content\"></slot>\n    <atlas-icon name=\"checkmark\" part=\"check\"></atlas-icon>\n  </div>\n  <div part=\"after\"></div>\n</template>\n";

const STEPPER_STEP_ATTRIBUTES = ['finished', 'current'];
let StepperStep = class StepperStep extends HTMLElement {
};
StepperStep = __decorate([
    Template(StepperStepTemplate)
], StepperStep);
if (!customElements.get('atlas-stepper-step'))
    customElements.define('atlas-stepper-step', StepperStep);

export { STEPPER_STEP_ATTRIBUTES, StepperStep };
