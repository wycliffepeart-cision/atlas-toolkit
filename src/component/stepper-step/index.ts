import { StepperStep } from './stepper-step';

if (!customElements.get('atlas-stepper-step')) customElements.define('atlas-stepper-step', StepperStep);
