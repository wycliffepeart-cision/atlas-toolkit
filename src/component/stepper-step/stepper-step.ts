import { Template } from '../../decorators/template';
import StepperStepTemplate from './stepper-step.html';

export const STEPPER_STEP_ATTRIBUTES = ['finished', 'current'];

@Template(StepperStepTemplate)
export class StepperStep extends HTMLElement {}

if (!customElements.get('atlas-stepper-step')) customElements.define('atlas-stepper-step', StepperStep);
