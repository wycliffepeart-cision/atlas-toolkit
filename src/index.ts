export * from './component/accordion/accordion';
export * from './component/alert/alert';
export * from './component/application-frame/application-frame';
export * from './component/button/button';
export * from './component/icon/icon';
export * from './component/button-set/button-set';
export * from './component/chip/chip';
export * from './component/chip-group/chip-group';
export * from './component/dialog/dialog';
export * from './component/feed/feed';
export * from './component/input-checkbox/input-checkbox';
export * from './component/input-chip-group/input-chip-group';
export * from './component/input-dropdown/input-dropdown';
export * from './component/input-search/input-search';
export * from './component/input-text/input-text';
export * from './component/input-textarea/input-textarea';
export * from './component/logo/logo';
export * from './component/main-nav/main-nav';
export * from './component/main-nav-item/main-nav-item';
export * from './component/number-display/number-display';
export * from './component/profile-control/profile-control';
export * from './component/snackbar/snackbar';
export * from './component/stepper/stepper';
export * from './component/stepper-step/stepper-step';
export * from './component/tile-board/tile-board';
export * from './component/tile-board-tile/tile-board-tile';
export * from './component/workflow/workflow';
export * from './component/workflow-step/workflow-step';
