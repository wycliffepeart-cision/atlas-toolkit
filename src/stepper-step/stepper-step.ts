import { Template } from '../decorators/template';
import StepperStepTemplate from './stepper-step.html';

export const STEPPER_STEP_ATTRIBUTES = ['finished', 'current'];

@Template(StepperStepTemplate)
export class StepperStep extends HTMLElement {}
