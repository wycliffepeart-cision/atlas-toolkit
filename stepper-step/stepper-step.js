import { _ as __decorate } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var StepperStepTemplate = "<template>\n  <div part=\"before\"></div>\n  <div part=\"step\">\n    <slot part=\"content\"></slot>\n    <atlas-icon name=\"checkmark\" part=\"check\"></atlas-icon>\n  </div>\n  <div part=\"after\"></div>\n</template>\n";

const STEPPER_STEP_ATTRIBUTES = ['finished', 'current'];
let StepperStep = class StepperStep extends HTMLElement {
};
StepperStep = __decorate([
    Template(StepperStepTemplate)
], StepperStep);

export { STEPPER_STEP_ATTRIBUTES, StepperStep };
