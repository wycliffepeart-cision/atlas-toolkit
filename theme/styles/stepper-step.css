@layer atlas {
	:root {
		--step-color-accent: var(--atlas-color-primary-b400);
		--step-color-base: var(--atlas-color-primary-n200);
		--step-check-color: var(--atlas-color-on-bg-white);

		--step-check-size: 1.125em;
		--step-size: 2.5em;
		--step-border-width: 1.5px;
		--step-bar-length: 3em;
		--step-spacer: 1ch;
		--step-bar-width: 1px;

		--step-bar-border: var(--step-bar-width) dashed var(--step-color-base);
		--step-finished-bar-border: var(--step-bar-width) solid var(--step-color-accent);

		--step-color: var(--step-color-base);
		--step-background-color: inherit;
		--step-border: var(--step-border-width) dashed var(--step-color-base);

		--step-current-color: var(--step-color-accent);
		--step-current-background-color: inherit;
		--step-current-border: var(--step-border-width) solid var(--step-color-accent);

		--step-finished-color: var(--step-check-color);
		--step-finished-background-color: var(--step-color-accent);
		--step-finished-border: 0;
	}

	atlas-stepper-step {
		display: flex;
		flex-wrap: nowrap;
		align-items: center;
	}

	atlas-stepper-step::part(check) {
		display: none;

		color: var(--step-check-color);

		width: 1em;
		height: 1em;
		font-size: var(--step-check-size);
	}

	atlas-stepper-step::part(step) {
		display: flex;
		margin-inline: auto;
		align-items: center;
		justify-content: center;

		width: var(--step-size);
		aspect-ratio: 1 / 1;
		border-radius: 100vw;
		overflow: hidden;
		white-space: nowrap;

		box-sizing: border-box;

		color: var(--step-color);
		background-color: var(--step-background-color);
		border: var(--step-border);
	}

	atlas-stepper-step::part(before) {
		display: block;
		width: var(--step-bar-length);
		height: 0;
		border-radius: 100vw;
		border: var(--step-bar-border);
		margin-inline: var(--step-spacer);
	}

	atlas-stepper-step::part(after) {
		display: none;
		width: var(--step-bar-length);
		height: 0;
		border-radius: 100vw;
		border: var(--step-bar-border);
		margin-inline: var(--step-spacer);
	}

	atlas-stepper-step[current]::part(step) {
		color: var(--step-current-color);
		background-color: var(--step-current-background-color);
		border: var(--step-current-border);
	}
	atlas-stepper-step[current]::part(before) {
		display: none;
	}
	atlas-stepper-step[current]::part(after) {
		display: none;
	}

	atlas-stepper-step[finished]::part(step) {
		color: var(--step-finished-color);
		background-color: var(--step-finished-background-color);
		border: var(--step-finished-border);
	}
	atlas-stepper-step[finished]::part(before) {
		display: none;
	}
	atlas-stepper-step[finished]::part(after) {
		display: block;

		background-color: var(--step-finished-bar-color);
		border: var(--step-finished-bar-border);
	}

	atlas-stepper-step[finished]::part(content) {
		display: none;
	}
	atlas-stepper-step[finished]::part(check) {
		display: block;
	}
}
