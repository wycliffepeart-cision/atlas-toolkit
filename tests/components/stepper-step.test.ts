import '../../src/stepper-step';

const STEPPER_STEP_TAG = 'atlas-stepper-step';
const KNOWN_ATTRIBUTES = ['finished', 'current'];

afterEach(() => {
  const elem = document.querySelector('body > *:first-child');
  elem?.remove();
});

describe('Stepper', () => {
  it('is correctly appended in the DOM', () => {
    const elem = document.createElement(STEPPER_STEP_TAG);
    document.body.appendChild(elem);

    (expect(elem) as any).toBeInTheDocument();
  });

  it('is matches snapshot', () => {
    const elem = document.createElement(STEPPER_STEP_TAG);
    document.body.appendChild(elem);

    (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
  });
});
