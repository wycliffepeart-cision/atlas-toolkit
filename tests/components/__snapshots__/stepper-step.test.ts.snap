// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper is matches snapshot 1`] = `
<div part="before">
</div>
<div part="step">
  <slot part="content">
  </slot>
  <atlas-icon name="checkmark"
              part="check"
  >
  </atlas-icon>
</div>
<div part="after">
</div>
`;
